src/browse.py

```diff
@@ -58,6 +58,17 @@ def match_strip(line, prefix):
         return (False, line)
     return (True, line[len(prefix):])
 
+html_escape_table = {
+    "&": "&amp;",
+    '"': "&quot;",
+    "'": "&apos;",
+    ">": "&gt;",
+    "<": "&lt;",
+    }
+
+def html_escape(text):
+    return "".join(html_escape_table.get(c,c) for c in text)
```

Comment:
cgi.escape doesn't escape a single quote.  Leaving single quotes in would be fine, unless someone introduced a single quoted HTML attribute in generate_html, in which case the links could break for any input or output file that contained a single quote.